src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -176,6 +180,29 @@ public boolean deleteFile(FireStoreFile fireStoreFile) {
     }
   }
 
+  public String signFile(
+      BillingProfileModel profileModel,
+      AzureStorageAccountResource storageAccountResource,
+      String url) {
+    BlobContainerClientFactory destinationClientFactory =
+        getTargetDataClientFactory(profileModel, storageAccountResource, true);
+
+    BlobUrlParts blobParts = BlobUrlParts.parse(url);
+    if (!blobParts.getAccountName().equals(storageAccountResource.getName())) {
+      throw new PdaoException(
+          String.format(
+              "Resource groups between metadata storage and request do not match: %s != %s",
+              blobParts.getAccountName(), storageAccountResource.getName()));
+    }
+    String blobName = blobParts.getBlobName();
+    BlobSasPermission permission = new BlobSasPermission().setReadPermission(true);
+    BlobSasTokenOptions blobSasTokenOptions =
+        new BlobSasTokenOptions(Duration.ofMinutes(15), permission, "");
```

Comment:
I'm not sure how the contentDisposition parameter gets used, but I've seen other places in the code use the name of the class that generates the sas token: https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/filedata/azure/util/BlobContainerCopier.java#L204

Reply:
Also from that same reference, maybe moving the duration into a variable would be good? Looks like the copier uses a 24 hour timeout?

Reply:
Agree it should be a variable.  Also, @saman, the content disposition is the field we can use to arbitrarily encode values.  The plan is to encode the user.  That leads me to think this variable should probably take a Authed user object